lib/pure/sugar.nim

```diff
@@ -231,9 +231,15 @@ macro capture*(locals: varargs[typed], body: untyped): untyped {.since: (1, 1).}
   let locals = if locals.len == 1 and locals[0].kind == nnkBracket: locals[0]
                else: locals
   for arg in locals:
-    if arg.strVal == "result":
+    var argStr = ""
+    case arg.kind
+    of nnkIdent, nnkSym: argStr = arg.strVal
+    of nnkHiddenDeref: argStr = arg[0].strVal
+    else:
+      error("The argument to be captured `" & arg.repr & "` is not a pure identifier.", arg)
```

Comment:
Yeah, I'm open to suggestions. I'm mainly not sure about how to phrase it without becoming overly long. I didn't write something like "complex expression", because I'm worried that some other node kind will also "look" like an identifier, but isn't. That could be more misleading than not.

Reply:
Does `arg.repr` print the node kind? It might be enough to add that, if nothing else.

Reply:
`repr` doesn't, no. Depending on the node kind it is even pretty confusing. But one could add an `$arg.kind`  to print the kind. As I alluded to in the opening though, I'm not sure if the stdlib anywhere prints this kind of information. I personally do it in my own macros though, as I find it useful.

Reply:
@Varriount I've extended the error message a bit.